src/php5/solr_functions_params.c

```diff
@@ -714,7 +714,9 @@ PHP_SOLR_API void solr_arg_list_param_value_fetch(solr_param_t *solr_param, solr
 		solr_string_append_solr_string(&tmp_buffer, &(current_ptr->contents.arg_list.value));
 
 		if (current_ptr->contents.arg_list.delimiter_override) {
-		    solr_string_appendc(&tmp_buffer, *current_ptr->contents.arg_list.delimiter_override);
+		    if (strlen(current_ptr->contents.arg_list.delimiter_override) > 0) {
+		        solr_string_appendc(&tmp_buffer, *current_ptr->contents.arg_list.delimiter_override);
+		    }
 		} else {
 		    solr_string_appendc(&tmp_buffer, separator);
 		}
```

Comment:
I think you need to combine these if-conditions, or add another else. first one is preferred, because second one just adds duplicate code.

1. combined
```c
		if (current_ptr->contents.arg_list.delimiter_override && strlen(current_ptr->contents.arg_list.delimiter_override) > 0) {
		    solr_string_appendc(&tmp_buffer, *current_ptr->contents.arg_list.delimiter_override);
		} else {
		    solr_string_appendc(&tmp_buffer, separator);
		}
```

2. else for second if

```c
		if (current_ptr->contents.arg_list.delimiter_override) {
		    if (strlen(current_ptr->contents.arg_list.delimiter_override) > 0) {
		        solr_string_appendc(&tmp_buffer, *current_ptr->contents.arg_list.delimiter_override);
		    } else {
		        solr_string_appendc(&tmp_buffer, separator);
		    }
		} else {
		    solr_string_appendc(&tmp_buffer, separator);
		}

```

Reply:
TNX @glensc!
Would you prefer a new cleanup - 3rd - commit or just amending the current one?

TIA,
Matteo

Reply:
While this does not apply fully, as changes are small and buffered in my memory, but:

As the review process is in order; you should add commits; it's rather annoying to review if the commits have been squashed in middle of review; i.e have to review again the whole codebase, can't know which is old and which is new change; you can tidy up the branch later when you find it's ready.

I personally use [git fixup](https://github.com/keis/git-fixup) tool aid `git rebase --autosquash`

Reply:
Hi @glensc,
pushed a 3rd commit, TNX for the suggestion/fix.

+1 for squashing all the commits into a single one, before merging it into the mainstream.

HTH,
Matteo